ecosystem/typescript/sdk_v2/src/types/index.ts

```diff
@@ -83,6 +85,7 @@ export type AptosRequest = {
   params?: Record<string, string | AnyNumber | boolean | undefined>;
   originMethod?: string;
   overrides?: ClientConfig;
+  isIndexer?: boolean;
```

Comment:
We may want an enum type rather than `isIndexer` to allow for future adaptability

Reply:
I agree it is a bit weird. At the end, the question whether `is indexer` depends on the url we use to query. I was trying to define `isAptosIndexer()` method in `aptos_config` and then call it in the `client/core.ts` file, but ended up with cycle dependency. 
What enum have you had in mind?

Reply:
Yeah seems less than ideal to add a param just to represent if it's the indexer even though that information is somewhat implicitly encoded in the url.

I'm guessing what Greg is suggesting is to pass in the `AptosApiType` and core.ts could gate on that to figure out how to handle the response?  It still would be semi-redundant but more adaptable than `isIndexer`

It would be nice if `aptosRequest` could derive the url and know how to handle the response via AptosApiType but you'd have to pass in the config I'm guessing which seems meh.  Not aware of the feasibility of that given the dependency graph.

I'll also suggest maybe instead adding an optional param to `AptosRequest` that is something like
`responseHandler?: (response: any) => AptosResponse<Req, Res>`
as a custom response handler that the query indexer can pass in?

This would allow for more adaptability to handle responses but idk if that bloats `AptosRequest` much.

just thought, I'm new here :)